<?xml version="1.0" ?>
<robot name="ajgar" xmlns:xacro="http://www.ros.org/wiki/xacro">

<xacro:include filename="$(find ajgar)/urdf/materials.xacro" />
<xacro:include filename="$(find ajgar)/urdf/ajgar_URDF.trans" />
<xacro:include filename="$(find ajgar)/urdf/ajgar_URDF.gazebo" />

<link name="world"/>

<link name="base_link">
  <inertial>
    <origin xyz="-0.00865082588772432 1.8113911902102217e-07 0.032368373908884494" rpy="0 0 0"/>
    <mass value="9.059758934586307"/>
    <inertia ixx="0.021583" iyy="0.026795" izz="0.042202" ixy="-0.0" iyz="-0.0" ixz="0.000269"/>
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="base_top">
  <inertial>
    <origin xyz="1.279003402099646e-05 -0.0022856514871783416 0.03289309768621855" rpy="0 0 0"/>
    <mass value="1.4850383611977547"/>
    <inertia ixx="0.002124" iyy="0.00176" izz="0.001747" ixy="-0.0" iyz="0.000118" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0.015013 -0.0 -0.06" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/base_top.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0.015013 -0.0 -0.06" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/base_top.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="shoulder">
  <inertial>
    <origin xyz="1.2461762721963782e-06 0.025783530669485587 0.11928257744286477" rpy="0 0 0"/>
    <mass value="2.8383656665985857"/>
    <inertia ixx="0.03026" iyy="0.030184" izz="0.00245" ixy="-0.0" iyz="0.000773" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0.015013 0.025 -0.119368" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/shoulder.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0.015013 0.025 -0.119368" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/shoulder.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="elbow">
  <inertial>
    <origin xyz="-0.0003292153850686253 0.026723336517713626 0.10654902767889785" rpy="0 0 0"/>
    <mass value="2.4072821039747727"/>
    <inertia ixx="0.026455" iyy="0.026758" izz="0.001672" ixy="-4e-06" iyz="0.000315" ixz="0.000123"/>
  </inertial>
  <visual>
    <origin xyz="0.015013 0.025 -0.377365" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/elbow.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0.015013 0.025 -0.377365" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/elbow.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="wrist">
  <inertial>
    <origin xyz="0.00920373509392226 -0.01118771679377522 0.025837849583827288" rpy="0 0 0"/>
    <mass value="0.34678950211541076"/>
    <inertia ixx="0.000595" iyy="0.000274" izz="0.000554" ixy="9.5e-05" iyz="7.4e-05" ixz="-6e-05"/>
  </inertial>
  <visual>
    <origin xyz="0.015013 -0.0 -0.644865" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/wrist.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0.015013 -0.0 -0.644865" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/wrist.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="suction">
  <inertial>
    <origin xyz="-8.479527444330004e-05 0.03652030172582285 8.898889773256702e-05" rpy="0 0 0"/>
    <mass value="0.6053245769138885"/>
    <inertia ixx="0.000476" iyy="0.000398" izz="0.000554" ixy="0.0" iyz="-0.0" ixz="1.7e-05"/>
  </inertial>
  <visual>
    <origin xyz="0.015013 0.037 -0.689865" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/suction.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0.015013 0.037 -0.689865" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/suction.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="end">
  <inertial>
    <origin xyz="-9.118561992635388e-08 3.576502294322381e-10 -0.00900061967229826" rpy="0 0 0"/>
    <mass value="0.07630173554088619"/>
    <inertia ixx="7e-06" iyy="5e-06" izz="8e-06" ixy="0.0" iyz="-0.0" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0.015013 -0.0 -0.714365" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/end.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0.015013 -0.0 -0.714365" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://ajgar/meshes/end.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<joint name="world_joint  " type="fixed">
  <parent link="world" />
  <child link="base_link" />
  <origin rpy="0 0 3.142" xyz="0.0 0.0 0.0" />
</joint>

<joint name="base_joint" type="revolute">
  <origin xyz="-0.015013 0.0 0.06" rpy="0 0 0"/>
  <parent link="base_link"/>
  <child link="base_top"/>
  <axis xyz="0.0 0.0 1.0"/>
  <limit upper="2.879793" lower="-2.879793" effort="100" velocity="100" />
</joint>

<joint name="shoulder_joint" type="revolute">
  <origin xyz="0.0 -0.025 0.059368" rpy="0 0 0"/>
  <parent link="base_top"/>
  <child link="shoulder"/>
  <axis xyz="0.0 -1.0 0.0"/>
  <limit upper="1.57" lower="-1.57" effort="100" velocity="100" />
</joint>

<joint name="arm_joint" type="revolute">
  <origin xyz="0.0 0.0 0.257997" rpy="0 0 0"/>
  <parent link="shoulder"/>
  <child link="elbow"/>
  <axis xyz="0.0 -1.0 -0.0"/>
  <limit upper="1.57" lower="-1.57" effort="100" velocity="100" />
</joint>

<joint name="forearm_joint" type="revolute">
  <origin xyz="0.0 0.025 0.2675" rpy="0 0 0"/>
  <parent link="elbow"/>
  <child link="wrist"/>
  <axis xyz="-0.0 0.0 1.0"/>
  <limit upper="2.879793" lower="-2.879793" effort="100" velocity="100" />
</joint>

<joint name="end_joint" type="revolute">
  <origin xyz="0.0 -0.037 0.045" rpy="0 0 0"/>
  <parent link="wrist"/>
  <child link="suction"/>
  <axis xyz="-0.0 1.0 -0.0"/>
  <limit upper="2.879793" lower="-2.879793" effort="100" velocity="100" />
</joint>

<joint name="suction_joint" type="revolute">
  <origin xyz="0.0 0.037 0.0245" rpy="0 0 0"/>
  <parent link="suction"/>
  <child link="end"/>
  <axis xyz="-0.0 -0.0 -1.0"/>
  <limit upper="2.879793" lower="-2.879793" effort="100" velocity="100" />
</joint>

</robot>
